x/distribution/keeper/store.go

```diff
@@ -23,93 +20,56 @@ func (k Keeper) GetDelegatorWithdrawAddr(ctx context.Context, delAddr sdk.AccAdd
 	return addr, err
 }
 
-// GetPreviousProposerConsAddr returns the proposer consensus address for the
-// current block.
-func (k Keeper) GetPreviousProposerConsAddr(ctx context.Context) (sdk.ConsAddress, error) {
-	store := k.storeService.OpenKVStore(ctx)
-	bz, err := store.Get(types.ProposerKey)
-	if err != nil {
-		return nil, err
-	}
-
-	if bz == nil {
-		return nil, errors.New("previous proposer not set")
-	}
+// historical reference count (used for testcases)
+func (k Keeper) GetValidatorHistoricalReferenceCount(ctx context.Context) (count uint64) {
+	iter, err := k.ValidatorHistoricalRewards.Iterate(
```

Comment:
I think here you could simply `.Walk` it. I'm not sure if the `Iterator`  does some savings on the fact that we don't call `Key`.

Reply:
I might move this one to where its used in the tests, no need to have it here.